libcxx/include/module.modulemap

```diff
@@ -1,2111 +1,1609 @@
+    header "__algorithm/ranges_remove.h"
+    header "__algorithm/ranges_replace_copy_if.h"
+    header "__algorithm/ranges_replace_copy.h"
+    header "__algorithm/ranges_replace_if.h"
+    header "__algorithm/ranges_replace.h"
+    header "__algorithm/ranges_reverse_copy.h"
+    header "__algorithm/ranges_reverse.h"
+    header "__algorithm/ranges_rotate_copy.h"
+    header "__algorithm/ranges_rotate.h"
+    header "__algorithm/ranges_sample.h"
+    header "__algorithm/ranges_search_n.h"
+    header "__algorithm/ranges_search.h"
+    header "__algorithm/ranges_set_difference.h"
+    header "__algorithm/ranges_set_intersection.h"
+    header "__algorithm/ranges_set_symmetric_difference.h"
+    header "__algorithm/ranges_set_union.h"
+    header "__algorithm/ranges_shuffle.h"
+    header "__algorithm/ranges_sort_heap.h"
+    header "__algorithm/ranges_sort.h"
+    header "__algorithm/ranges_stable_partition.h"
+    header "__algorithm/ranges_stable_sort.h"
+    header "__algorithm/ranges_starts_with.h"
+    header "__algorithm/ranges_swap_ranges.h"
+    header "__algorithm/ranges_transform.h"
+    header "__algorithm/ranges_unique_copy.h"
+    header "__algorithm/ranges_unique.h"
+    header "__algorithm/ranges_upper_bound.h"
+    header "__algorithm/remove_copy_if.h"
+    header "__algorithm/remove_copy.h"
+    header "__algorithm/remove_if.h"
+    header "__algorithm/remove.h"
+    header "__algorithm/replace_copy_if.h"
+    header "__algorithm/replace_copy.h"
+    header "__algorithm/replace_if.h"
+    header "__algorithm/replace.h"
+    header "__algorithm/reverse_copy.h"
+    header "__algorithm/reverse.h"
+    header "__algorithm/rotate_copy.h"
+    header "__algorithm/rotate.h"
+    header "__algorithm/sample.h"
+    header "__algorithm/search_n.h"
+    header "__algorithm/search.h"
+    header "__algorithm/set_difference.h"
+    header "__algorithm/set_intersection.h"
+    header "__algorithm/set_symmetric_difference.h"
+    header "__algorithm/set_union.h"
+    header "__algorithm/shift_left.h"
+    header "__algorithm/shift_right.h"
+    header "__algorithm/shuffle.h"
+    header "__algorithm/sift_down.h"
+    header "__algorithm/simd_utils.h"
+    header "__algorithm/sort_heap.h"
+    header "__algorithm/sort.h"
+    header "__algorithm/stable_partition.h"
+    header "__algorithm/stable_sort.h"
+    header "__algorithm/swap_ranges.h"
+    header "__algorithm/three_way_comp_ref_type.h"
+    header "__algorithm/transform.h"
+    header "__algorithm/uniform_random_bit_generator_adaptor.h"
+    header "__algorithm/unique_copy.h"
+    header "__algorithm/unique.h"
+    header "__algorithm/unwrap_iter.h"
+    header "__algorithm/unwrap_range.h"
+    header "__algorithm/upper_bound.h"
+
+    export debug_utils.strict_weak_ordering_check
+    export functional.ranges_operations
+    export iterator.aliasing_iterator
+    export iterator.iterator_traits
+    export type_traits.is_trivially_copyable
+    export utility.pair
```

Comment:
I've run into weird issues when I do `export specific_thing` and `export *` at the same time. In theory the specific exports are redundant and already included by `*`, but in practice it's caused some random behavior where not everything gets exported and it doesn't even seem related to the specific exports. If we're need to export `*` then we do already get everything, and so we should remove or at least comment out the specific exports.

Reply:
Re-exports are useful when two header files are dependent on each other and we do not know which one is included first.

Reply:
Sure, but `export *` will export everything the module imported, and `export whatever` doesn't do anything if you didn't import `whatever`. So if you have `export *`, that's all you need, and in fact adding specific modules will mess clang up.